### PR TITLE
Issue #15 fix: User update (patch) is not working for resource `app=users&resource=user`

### PR DESCRIPTION
**Pull request by manojLondhe**
- [x] Update, Delete - Add missing lang constants 
- [x] Update - Issue #15 fix: User update (patch) is not working for resource `app=users&resource=user`
- [x] Get - Issue #5 fix: user.php returning user's password
- [x] Update - Ignore password field if not set to avoid warning on bind()
- [x] Improve variable names
- [x] Improve comments

### CODE REVIEW

File: src/users/user.php [hunk 1]
```diff
@@ -112,11 +105,12 @@ public function post()
 			}
 			else
 			{
-				if ($fidentifier)
+				// Forced user creation
+				if ($fIdentifier)
```
Comment (coolbung) [inline src/users/user.php:109]: Combine this with the else above into an elseif condition ?

File: src/users/user.php [hunk 2]
```diff
@@ -28,29 +28,22 @@ class UsersApiResourceUser extends ApiResource
...
+		$app            = JFactory::getApplication();
+		$params         = JComponentHelper::getParams("com_users");
+		$formData       = $app->input->getArray();
+		$userIdentifier = $app->input->get('id', 0, 'string');
+		$xIdentifier    = $app->input->server->get('HTTP_X_IDENTIFIER');
+		$fIdentifier    = $app->input->server->get('HTTP_FORCECREATE');
 
 		// Get current logged in user.
-		$my = JFactory::getUser();
+		$me = JFactory::getUser();
```
Comment (coolbung) [inline src/users/user.php:39]: The user object is already available in `$this->plugin->get('user')` can you confirm it is ok to use either ?

File: src/users/user.php [hunk 3]
```diff
@@ -28,29 +28,22 @@ class UsersApiResourceUser extends ApiResource
...
+
+		// Check if $userIdentifier is not set - POST / CREATE user case
 
-		// Check if $userIdentifier is not set
 		if (empty($userIdentifier))
 		{
-			if ($formData['password'] == '')
+			// Validate required fields
+			if ($formData['username'] == '' || $formData['name'] == '' || $formData['email'] == '' || $formData['password'] == '')
```
Comment (vaivk369) [inline src/users/user.php:46]: We can create user without passing the password in joomla.  
@manojLondhe

File: src/users/user.php [hunk 4]
```diff
@@ -112,11 +105,12 @@ public function post()
 			}
 			else
 			{
-				if ($fidentifier)
+				// Forced user creation
+				if ($fIdentifier)
 				{
 					$user = new JUser;
```
Comment (vaivk369) [inline src/users/user.php:111]: If id is provided, and is exist then user should get updated
@manojLondhe @coolbung

**Comment by manojLondhe** [comment]
closing in favor of https://github.com/techjoomla/plg_api_users/pull/17

**Pull request closed without merging**